go/vt/vttablet/tabletmanager/vreplication/vplayer.go

```diff
@@ -133,6 +138,25 @@ func (vp *vplayer) play(ctx context.Context) error {
 	return vp.fetchAndApply(ctx)
 }
 
+// vplayer starts with fk checks off. updateFKCheck will check if the fk checks value has changed from what it is currently
+// set to and update it if it has. This is done to avoid setting the fk checks value for every row event.
+func (vp *vplayer) updateFKCheck(ctx context.Context, flags2 uint32) error {
+	// The 2nd bit (least significant) of the flags attribute of the binlog row event
+	// is set to 1 if foreign key checks are disabled.
+	enableFK := true
+	if flags2&NoForeignKeyCheckFlagBitmask == NoForeignKeyCheckFlagBitmask {
+		enableFK = false
+	}
+	if vp.foreignKeyCheck == enableFK {
+		// if not changed, return
+		return nil
+	}
+	vp.foreignKeyCheck = enableFK
+	log.Infof("Setting foreign_key_checks to %v", enableFK)
+	_, err := vp.vr.dbClient.ExecuteWithRetry(ctx, "set foreign_key_checks="+strconv.FormatBool(enableFK))
```

Comment:
It's good to get into the habit of being explicit about variable scope -- even though this variable ONLY exists in the session scope today.